core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -2986,6 +2989,198 @@ object ZManaged extends ZManagedPlatformSpecific {
+   * to depend on `Input`.
+   */
+  sealed trait ZManagedConstructor[Input] {
+
+    /**
+     * The type of the `ZManaged` value.
+     */
+    type Out
+
+    /**
+     * Constructs a `ZManaged` value from the specified input.
+     */
+    def make(input: => Input): Out
+  }
+
+  object ZManagedConstructor extends ZManagedConstructorLowPriority1 {
+
+    /**
+     * Constructs a `ZManaged[Any, E, A]` from an `Either[E, A]`.
+     */
+    implicit def AutoCloseableConstructor[R, E, A <: AutoCloseable]: WithOut[ZIO[R, E, A], ZManaged[R, E, A]] =
+      new ZManagedConstructor[ZIO[R, E, A]] {
+        type Out = ZManaged[R, E, A]
+        def make(input: => ZIO[R, E, A]): ZManaged[R, E, A] =
+          ZManaged.fromAutoCloseable(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[Any, E, A]` from an `Either[E, A]`.
+     */
+    implicit def EitherConstructor[E, A]: WithOut[Either[E, A], ZManaged[Any, E, A]] =
+      new ZManagedConstructor[Either[E, A]] {
+        type Out = ZManaged[Any, E, A]
+        def make(input: => Either[E, A]): ZManaged[Any, E, A] =
+          ZManaged.fromEither(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[Any, E, A]]` from an `Either[E, A]`.
+     */
+    implicit def EitherLeftConstructor[E, A]: WithOut[Left[E, A], ZManaged[Any, E, A]] =
+      new ZManagedConstructor[Left[E, A]] {
+        type Out = ZManaged[Any, E, A]
+        def make(input: => Left[E, A]): ZManaged[Any, E, A] =
+          ZManaged.fromEither(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[Any, E, A]` from an `Either[E, A]`.
+     */
+    implicit def EitherRightConstructor[E, A]: WithOut[Right[E, A], ZManaged[Any, E, A]] =
+      new ZManagedConstructor[Right[E, A]] {
+        type Out = ZManaged[Any, E, A]
+        def make(input: => Right[E, A]): ZManaged[Any, E, A] =
+          ZManaged.fromEither(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[R, E, A]` from a function `R => IO[E, A]`.
+     */
+    implicit def FunctionManagedConstructor[R, E, A]: WithOut[R => ZManaged[Any, E, A], ZManaged[R, E, A]] =
+      new ZManagedConstructor[R => ZManaged[Any, E, A]] {
+        type Out = ZManaged[R, E, A]
+        def make(input: => (R => ZManaged[Any, E, A])): ZManaged[R, E, A] =
+          ZManaged.fromFunctionManaged(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[Any, Option[Nothing], A]` from an `Option[A]`.
+     */
+    implicit def OptionConstructor[A]: WithOut[Option[A], ZManaged[Any, Option[Nothing], A]] =
+      new ZManagedConstructor[Option[A]] {
+        type Out = ZManaged[Any, Option[Nothing], A]
+        def make(input: => Option[A]): ZManaged[Any, Option[Nothing], A] =
+          ZManaged.fromOption(input)
+      }
+
+    /**
+     * Constructs a `ZManaged[Any, Option[Nothing], A]` from a `None`.
+     */
+    implicit def OptionNoneConstructor[A]: WithOut[None.type, ZManaged[Any, Option[Nothing], A]] =
```

Comment:
Same as with the ZIO instance:
```suggestion
    implicit val OptionNoneConstructor: WithOut[None.type, ZManaged[Any, Option[Nothing], Nothing]] =
```

Reply:
Done.